documentation/docs/mock-apps/components/render-check-list.md

```diff
@@ -48,24 +47,118 @@ The RenderCheckList component is responsible for rendering a list of items with
   },
-  "9876543210": {
-    // Imported data
+  "data2.json": {
+    "foo": "bar" // Imported data
+  }
+}
+```
+
+2. If the imported or scanned data type is a `VerifiableCredential`, the data will be returned as an object with the key as the file name or URL of QRcode. The imported or scanned VC will be verified based on the `vcOptions` provided in the [Import Button](./import-button) or [QR Code Scanner Dialog Button](./qr-code-scanner-dialog-button) component.
+
+```json
+{
+  "VC1.json": {
+    "@context": [], // Imported VC
+    "type": ["VerifiableCredential"]
+  },
+  "VC2.json": {
+    "@context": [], // Imported VC
+    "type": ["VerifiableCredential"]
+  }
+}
+```
+
+3. If the imported or scanned data type is a `VerifiableCredential` and the VC is an enveloped VC, the data will be returned as an object with the key as the file name or URL of QRcode. The imported or scanned VC will be verified and decoded based on the `vcOptions` provided in the [Import Button](./import-button) or [QR Code Scanner Dialog Button](./qr-code-scanner-dialog-button) component. The decoded VC will return the same level path as the original VC.
+   For example, if the original VC was imported as the whole object of the file, the decoded VC will be returned together with the original VC in an object like below:
+
+```json
+// imported VC file
+{
+  "@context": ["https://www.w3.org/ns/credentials/v2"],
+  "id": "data:application/vc-ld+jwt,jwt",
+  "type": "EnvelopedVerifiableCredential"
+}
+```
+
+```json
+// return data
+{
+  "VC1.json": {
+    "vc": {
+      "@context": ["https://www.w3.org/ns/credentials/v2"],
+      "id": "data:application/vc-ld+jwt,jwt",
+      "type": "EnvelopedVerifiableCredential"
+    },
+    "decodedEnvelopedVC": {
+      "@context": ["https://www.w3.org/ns/credentials/v2"],
+      "type": ["VerifiableCredential"],
+      "credentialSubject": {
+        "id": "did:example:123",
+        "name": "Alice"
+      }
+    }
+  }
+}
+```
+
+But when imported data has `credentialPath` in `vcOptions`, the decoded VC will be returned with the path specified in `credentialPath` together with the original VC in an object like below:
+
+```json
+// vcOptions
+{
+  "credentialPath": "/vc"
+}
+```
+
+```json
+// imported VC file
+{
+  "vc": {
+    "@context": ["https://www.w3.org/ns/credentials/v2"],
+    "id": "data:application/vc-ld+jwt,jwt",
+    "type": "EnvelopedVerifiableCredential"
+  },
+  "linkResolver": "https://example.com"
+}
+```
+
+```json
+// return data
+{
+  "VC1.json": {
+    "vc": {
+      "vc": {
+        "@context": ["https://www.w3.org/ns/credentials/v2"],
+        "id": "data:application/vc-ld+jwt,jwt",
+        "type": "EnvelopedVerifiableCredential"
+      },
+      "decodedEnvelopedVC": {
+        "@context": ["https://www.w3.org/ns/credentials/v2"],
+        "type": ["VerifiableCredential"],
+        "credentialSubject": {
+          "id": "did:example:123",
+          "name": "Alice"
+        }
+      }
```

Comment:
Should we add a brief description of the `decodedEnvelopedVC` field here? I think this document would be better if we explained how it is generated.